admin/acceptance.py

```diff
@@ -671,6 +673,37 @@ def error(failure):
         d.addErrback(error)
         return d
 
+    def _provision_with_retry(self, reactor, name, result):
+        def provision(node):
+            print "created node", name
+            print "provisioning..."
+            self.nodes.append(node)
+            d = self._provision_node(reactor, node)
+
+            def provisioning_failed(failure):
+                print "provisioning failed", name
+                # Destroy a node if we failed to provision it.
+                self._destroy_node(node)
+                return failure
+
+            # Make sure to return the node.
+            d.addCallbacks(lambda _: node, errback=provisioning_failed)
+            return d
+
+        result.addCallback(provision)
+
+        # If the initial bulk creation failed for this node for whatever
+        # reason or if its provisioning failed, then beging the process
+        # for this node from the very start.
+        def retry_on_error(failure):
+            print "bulk creation failed", name
```

Comment:
Same as above. Eliot is already being used for some outputs in this file, and we use it as well in benchmark code, it may be a good thing to be consistent with what we use - either print or Eliot, but using both doesn't look very consistent.